### DISCUSSION TITLE
User Editable Peer Version Restrictions

### DISCUSSION THREAD

**Discussion by hwfanatic** [Chia-Network/chia-blockchain#15669] [Ideas]
With 1.8.2, the chia client has begun removing peers that are on old versions of the chia client.
This request is to expose the settings for the user to edit and control what versions of the chia client that peers must have to remain connected.

This can initially be added as a config.yaml entry (potentially an entry that needs to be added by the user) then it would be great to expose in cli and GUI settings.

Original --------
Give us the ability to set the minimum required version for full node connections, OR devise some other way to prevent connections with nodes that are stuck on blocks 3228*** and 3455***.

It is getting really old and tiresome to have to manually clean the list all the time.

**Comment by BrandtH22** [comment]
Hey @hwfanatic , you should notice that in 1.8.2 many peers will start to be banned for their software version.
There is no easy method to set this yourself as of yet and that likely will not be available until a future release (not yet determined when or if unfortunately).

Does 1.8.2 keep your peer list cleaner now?

**Comment by hwfanatic** [comment]
@BrandtH22 I don't think 1.8.2 is banning any peers or at least I haven't noticed that my peer list is any cleaner now. The number of peers with outdated software has slightly decreased in the past month, but I can only presume this has happened organically.

**Reply by BrandtH22** [reply]
Hey @hwfanatic, thank you for the response and I am glad that the peer list is seeming to be more cleaned up.
That could well be an effect of the versioning control within 1.8.2, you could review the log files to verify as there is a new message that would indicate "banning peer xxx.xxx.xxxx maybe they are running an old version of the client" .

It is not a very strict filter currently but it should have the effect of dropping any peers that are on old enough versions to run into issues with forks that have already been undergone by the chain as a whole (those old pre 1.4 peers primarily for now).

Did you want to update this feature request to "Provide user editable version minimums for full node peers", keep it as is, or something else?

**Reply by hwfanatic** [reply]
@BrandtH22 Okay, I can see those messages in the log. I guess the filter is working, but it is still letting some peers through. Maybe the filter should be looked at and possibly tightened, or there could be some other issue at hand.

If this is already in the works, we can close the feature request.

**Reply by BrandtH22** [reply]
Good afternoon @hwfanatic, to ensure that we keep eyes on this request I am going to update the title and tags to reflect a feature request to allow the user to edit the version restriction. 
It likely will not be available in the next release but once we have the engineering hours to develop out the features we will be able to update the info here at that time.

In the meantime feel free to add any additional information to this discussion or reach out with any followup questions :+1:
